cmake/AzAutoGen.py

```diff
@@ -37,7 +37,15 @@ def __init__(self, targetName, cacheDir, outputDir, projectDir, inputFiles, expa
         self.pythonPaths = pythonPaths
 
 def SanitizeTargetName(targetName):
-    return targetName.replace('.', '').replace('::', '')
+    targetNameList = list(targetName)
+    firstAlphabetIndex = -1
+    for i, ch in enumerate(targetNameList):
+        if ch.isalpha() and firstAlphabetIndex == -1:
+            firstAlphabetIndex = i
+        if not ch.isalnum():
+            targetNameList[i] = ""
+
+    return ''.join(targetNameList[firstAlphabetIndex:]) if 0 <= firstAlphabetIndex < len(targetNameList) else ""
```

Comment:
If the goal here is to limit the target name to alphanumeric characters, then let's use a regular expression:
```suggestion
    return re.sub(r'[^\w]', '', name)
```

Reply:
also need non leading number

Reply:
That would be:
```python
return re.sub(r'[^\w]', '', name.lstrip('0123456789'))
```

Reply:
cool, looks more fit in python, will give a try